teuthology/test/test_misc.py

```diff
@@ -150,3 +150,15 @@ def test_missing_b_deep_merge(self):
     def test_invalid_b_deep_merge(self):
         with pytest.raises(AssertionError):
             misc.deep_merge({"a": "b"}, "invalid")
+
+
+class TestIsInDict(object):
+    def test_is_in_dict(self):
+        assert misc.is_in_dict('a', 'foo', {'a':'foo', 'b':'bar'})
+        assert misc.is_in_dict(
```

Comment:
This ought to be four separate test methods, so that if one fails it's more clear which one failed.